From 875fbbd92695397bfc83d1cd5fdd4094e1d50199 Mon Sep 17 00:00:00 2001
From: Larry Gritz <lg@larrygritz.com>
Date: Mon, 28 Dec 2015 11:46:07 -0800
Subject: [PATCH] Python ImageCache binding fixes -- disable broken calls

Some of these calls (thankfully undocumented and presumably unused)
are horribly broken. They compiled before, but with new Boost 1.60
they don't even compile properly. So just comment them out on this
obsolete branch. They are fully fixed in RB-1.6 and beyond.

--- a/src/python/py_imagecache.cpp
+++ b/src/python/py_imagecache.cpp
@@ -199,23 +199,24 @@ void declare_imagecache()
         .def("destroy", &ImageCacheWrap::destroy)
         .staticmethod("destroy")
         .def("clear", &ImageCacheWrap::clear)
-        .def("attribute", &ImageCacheWrap::attribute)
+        // .def("attribute", &ImageCacheWrap::attribute)
         .def("attribute", &ImageCacheWrap::attribute_int)
         .def("attribute", &ImageCacheWrap::attribute_float)
-        .def("attribute", &ImageCacheWrap::attribute_double)
-        .def("attribute", &ImageCacheWrap::attribute_char)
+        // .def("attribute", &ImageCacheWrap::attribute_double)
+        // .def("attribute", &ImageCacheWrap::attribute_char)
         .def("attribute", &ImageCacheWrap::attribute_string)
-        .def("getattribute", &ImageCacheWrap::attribute)
+        // .def("getattribute", &ImageCacheWrap::attribute)
         .def("getattribute", &ImageCacheWrap::getattribute_int)
         .def("getattribute", &ImageCacheWrap::getattribute_float)
-        .def("getattribute", &ImageCacheWrap::getattribute_double)
-        .def("getattribute", &ImageCacheWrap::getattribute_char)
+        // .def("getattribute", &ImageCacheWrap::getattribute_double)
+        // .def("getattribute", &ImageCacheWrap::getattribute_char)
         .def("getattribute", &ImageCacheWrap::getattribute_string)
-        .def("resolve_filename", &ImageCacheWrap::resolve_filename)
-        .def("get_image_info", &ImageCacheWrap::get_image_info)
-        .def("get_image_info", &ImageCacheWrap::get_image_info_old)
+        // .def("get_image_info", &ImageCacheWrap::get_image_info)
+        // .def("get_image_info", &ImageCacheWrap::get_image_info_old)
         .def("get_imagespec", &ImageCacheWrap::get_imagespec)
-        .def("get_pixels", &ImageCacheWrap::get_pixels)
+        // .def("get_pixels", &ImageCacheWrap::get_pixels)
+        .def("resolve_filename", &ImageCacheWrap::resolve_filename)
+
 //      .def("get_tile", &ImageCacheWrap::get_tile)
 //      .def("release_tile", &ImageCacheWrap::release_tile)
 //      .def("tile_pixels", &ImageCacheWrap::tile_pixels)
